src/app/atoms/co-social-button/co-social-button.component.html

```diff
@@ -0,0 +1,24 @@
+<button class="co-social-button">
+  <span class="co-social-button__icon-container"
+        *ngIf="socialNetwork == 'facebook'">
+    <svg style="width:24px;height:24px" viewBox="0 0 24 24">
+        <path fill="#4267b2" d="M17,2V2H17V6H15C14.31,6 14,6.81 14,7.5V10H14L17,10V14H14V22H10V14H7V10H10V6A4,4 0 0,1 14,2H17Z" />
+    </svg>
+  </span>
+  <span class="co-button__title" 
+        *ngIf="socialNetwork == 'facebook'"
+        [ngClass]="{'co-social-button__text-button--facebook': socialNetwork == 'facebook'}">
+    Entrar com o Facebook
+  </span>
+  <span class="co-social-button__icon-container"
+        *ngIf="socialNetwork == 'google'">
+    <svg style="width:24px;height:24px" viewBox="0 0 24 24">
+      <path fill="#ea4335" d="M21.35,11.1H12.18V13.83H18.69C18.36,17.64 15.19,19.27 12.19,19.27C8.36,19.27 5,16.25 5,12C5,7.9 8.2,4.73 12.2,4.73C15.29,4.73 17.1,6.7 17.1,6.7L19,4.72C19,4.72 16.56,2 12.1,2C6.42,2 2.03,6.8 2.03,12C2.03,17.05 6.16,22 12.25,22C17.6,22 21.5,18.33 21.5,12.91C21.5,11.76 21.35,11.1 21.35,11.1V11.1Z" />
+    </svg>
+  </span>
+  <span class="co-button__title" 
+        *ngIf="socialNetwork == 'google'"
+        [ngClass]="{'co-social-button__text-button--google': socialNetwork == 'google'}">
+    Entrar com o Google
+  </span>
+</button>
```

Comment:
Você não precisa de um span para cada título. Você pode fazer só um assim: Entrar com o {{socialNetwork}}. Dessa forma é só passar o nome da rede social com a primeira letra maiúscula quando for chamar o componente.

Reply:
Ei você usa o ngClass para mudar a cor de acordo com a rede social.